### PR TITLE
Cloud init: remove unused function

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
Moves a function to a test file as it is not needed in production code.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloud-init/cloud-init_test.go [hunk 1]
```diff
@@ -266,15 +266,6 @@ var _ = Describe("CloudInit", func() {
 				Expect(err).To(HaveOccurred())
 				Expect(timedOut).To(BeTrue())
 			})
-
-			Context("when local data does not exist", func() {
```
Comment (acardace) [inline pkg/cloud-init/cloud-init_test.go:270]: I guess this test is just bogus?

Comment (xpivarc) [reply pkg/cloud-init/cloud-init_test.go:270]: Not sure  how it got to be alive

**Pull request merged**